src/with_position.rs

```diff
@@ -81,6 +81,34 @@ impl<I: Iterator> Iterator for WithPosition<I> {
     fn size_hint(&self) -> (usize, Option<usize>) {
         self.peekable.size_hint()
     }
+
+    fn fold<B, F>(mut self, mut init: B, mut f: F) -> B
+    where
+        F: FnMut(B, Self::Item) -> B,
+    {
+        if !self.handled_first {
+            // Haven't seen the first item yet, and there might be one to give.
+            self.handled_first = true;
```

Comment:
Do we actually need this? `self` is consumed and `handled_first` seems unused afterwards.